acareeg/eegip.py

```diff
@@ -171,53 +175,122 @@ def process_events_london_resting_state(raw, age, tmax=1):
 
         if len(rst0) == 0 and len(rst1) == 0:
             return None
+        assert(len(rst0) == len(rst1))
 
-        annot_sample = np.concatenate([np.arange(start, stop - 0.999, 1.0) for start, stop in zip(rst0, rst1)])
-        annot_sample = (annot_sample * freq).astype(int)
+        # Keep only trials that are at least 10 s long.
+        # All trials are defined by Rst0-Rst1 pair of events.
+        annot_sample = np.concatenate([np.arange(start*freq, (stop-tmax)*freq, tmax*freq)
+                                       for start, stop in zip(rst0, rst1)
+                                       if stop - start > 10.0]).astype(int)
         annot_id = [1] * len(annot_sample)
 
     elif age == 12:
-        annots = [a for a in raw.annotations if a["description"] in ["eeg1", "eeg2", "eeg3"]]
-        if len(annots) == 0:
+        """
+         For London 12m, resting-state data are marked only by Rst0 markers at the begining of
+         every trials. Trials and consecutives, so we can use the following Rst0 maker as being the 
+         end of the current trials. The last trials is ended with the last occurence of a SWIR or SWEN
+         event. 
+         
+         Trials shorter than 10s are rejected. Trials longer than 50s are truncated at 50s.
+         
+         Events eeg1, eeg2 and eeg3, when present, are at the same time as the Rst0 events. They 
+         are used to set the type of event (type of video that was presented). If no eegx event
+         is available for a Rst0, as type eeg4 is set, which indicate a resting-state event using
+         an unknown video.
+        """
+
+        onsets = [annot["onset"] for annot in raw.annotations if annot["description"] == "Rst0"]
+        if len(onsets) == 0:
             return None
 
-        if raw.annotations[-1]["onset"] > annots[-1]["onset"]:
-            end = np.min([raw.annotations[-1]["onset"], annots[-1]["onset"] + 50.])
```

Comment:
end - the last raw.annotations onset plus 50 seconds? I don't understand the ", annots[-1]["onset"].."

Reply:
Idem. Old code. Was replaced because it was not the best approach. The important is that the new code do what we want it to do.